csharp/src/Google.Protobuf/JsonFormatter.cs

```diff
@@ -758,31 +764,42 @@ public sealed class Settings
             /// <summary>
             /// Default settings, as used by <see cref="JsonFormatter.Default"/>
             /// </summary>
-            public static Settings Default { get; }
+            public static Settings Default { get; } = new Settings();
 
-            // Workaround for the Mono compiler complaining about XML comments not being on
-            // valid language elements.
-            static Settings()
+            /// <summary>
+            /// Creates a new <see cref="Settings"/> object.
+            /// </summary>
+            public Settings() : this(false, TypeRegistry.Empty)
             {
-                Default = new Settings(false);
             }
 
             /// <summary>
             /// Whether fields whose values are the default for the field type (e.g. 0 for integers)
             /// should be formatted (true) or omitted (false).
             /// </summary>
-            public bool FormatDefaultValues { get; }
+            public bool FormatDefaultValues { get; set; }
 
+            private TypeRegistry typeRegistry_;
             /// <summary>
             /// The type registry used to format <see cref="Any"/> messages.
             /// </summary>
-            public TypeRegistry TypeRegistry { get; }
+            public TypeRegistry TypeRegistry {
+                get { return typeRegistry_ ; }
+                set {
+                    typeRegistry_ = ProtoPreconditions.CheckNotNull(value, nameof(value));
+                }
+            }
+
+            /// <summary>
+            /// Whether to format enums as ints. Defaults to false.
+            /// </summary>
+            public bool FormatEnumAsInt { get; set;}
```

Comment:
I'd expand `Int` to `Integer` here; in .NET APIs normally you'd refer to `Int32` if you mean that specified type, or `Integer` to refer to integers in general.

Also, to be consistent with `FormatDefaultValues`, I'd be tempted to use the plural again: `FormatEnumsAsIntegers`